### PR TITLE
Convert mappings extension to storage unit

### PR DESCRIPTION
**Pull request by NoComment1105**
Re-done this time. Should be fully functioning.

### CODE REVIEW

File: extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/stroage/MappingsConfig.kt
Comment (Galarzaa90) [inline extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/stroage/MappingsConfig.kt]: Did you mean to type `storage`? 🤔

Comment (NoComment1105) [reply extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/stroage/MappingsConfig.kt]: No.

Comment (NoComment1105) [reply extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/stroage/MappingsConfig.kt]: Wait ignore me I thought you were on about something else

Comment (NoComment1105) [reply extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/stroage/MappingsConfig.kt]: Fixed that

**Comment by gdude2002** [comment]
I'm unsure what specifically to do with this one.

The mappings extension is useful for sure, but it's been impossible for me to maintain it for a long time, and @sschr15 seems like he's too busy to do much with it these days.

Additionally, the storage unit system needs a major rework at some point, something which I'm very unlikely to be able to do any time soon.

There is a conflict to resolve here, but I'm left feeling like I didn't give you the right brief to solve this properly - but I don't have the spoons to do a proper review right now anyway, so I'm not sure.

**Comment by NoComment1105** [comment]
Hm, Ok then. I'll resolve the conflict, then let me know when you have appropriate spoons

File: extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/MappingsExtension.kt
Comment (sschr15) [inline extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/MappingsExtension.kt]: Why the extra suppressions? These generally are unpleasant code features that usually should be covered on a case-by-case basis

File: extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/MappingsExtension.kt
Comment (sschr15) [inline extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/MappingsExtension.kt]: Unless I'm missing something and/or Dokka's checking has changed since the last time this class was modified, this seems to be the only location of a `MagicNumber` smell in the diff.

**Review by sschr15 — APPROVED** [review]
Overall looks good. Fantastic that I didn't realize I had been mentioned for a bit of code I've done things with

**Comment by gdude2002** [comment]
It's unfortunate that this has gone so long until you managed to get a review in - but when the merge conflicts are solved, things are re-tested, and you're happy with it, we can probably merge @sschr15

**Review by sschr15 — APPROVED** [review]

File: extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/MappingsExtension.kt
Comment (sschr15) [inline extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/MappingsExtension.kt]: This line crashes because `SlashCommand.guildId` is null

**Review by sschr15 — CHANGES_REQUESTED** [review]

File: extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/MappingsExtension.kt
Comment (NoComment1105) [reply extra-modules/extra-mappings/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/mappings/MappingsExtension.kt]: woe. What fix did you manage to fixed it with?

**Review by sschr15 — APPROVED** [review]

**Comment by NoComment1105** [comment]
@gdude2002 Could this have a final review please? sschr15 has approved it

**Review by gdude2002 — DISMISSED** [review]
I can't find much to criticise here - but as always, I need confirmation that these changes were tested.

**Comment by sschr15** [comment]
Unless the merge commit broke something, I did testing before sending over the patch that was implemented

**Review by gdude2002 — DISMISSED** [review]

**Comment by gdude2002** [comment]
Alright, well then I'll merge it when you've got it to pass the linter.

image

**Comment by gdude2002** [comment]
That was an odd one, wasn't it?

Anyway, thanks, I think we finally made it!

**Pull request merged**
